test/server/admin/admin_test.cc

```diff
@@ -248,6 +489,91 @@ TEST_P(AdminInstanceTest, ConfigDumpFiltersByResource) {
   EXPECT_EQ(expected_json, output);
 }
 
+TEST_P(AdminInstanceTest, ConfigDumpWithEndpointFiltersByResource) {
+  Upstream::ClusterManager::ClusterInfoMap cluster_map;
+  ON_CALL(server_.cluster_manager_, clusters()).WillByDefault(ReturnPointee(&cluster_map));
+
+  NiceMock<Upstream::MockClusterMockPrioritySet> cluster;
+  cluster_map.emplace(cluster.info_->name_, cluster);
+
+  ON_CALL(*cluster.info_, addedViaApi()).WillByDefault(Return(false));
+
+  Upstream::MockHostSet* host_set = cluster.priority_set_.getMockHostSet(0);
+  auto host = std::make_shared<NiceMock<Upstream::MockHost>>();
+
+  envoy::config::core::v3::Locality locality;
+  ON_CALL(*host, locality()).WillByDefault(ReturnRef(locality));
+
+  host_set->hosts_.emplace_back(host);
+  Network::Address::InstanceConstSharedPtr address =
+      Network::Utility::resolveUrl("tcp://1.2.3.4:80");
+  ON_CALL(*host, address()).WillByDefault(Return(address));
+  const std::string hostname = "foo.com";
+  ON_CALL(*host, hostname()).WillByDefault(ReturnRef(hostname));
+
+  auto metadata = std::make_shared<envoy::config::core::v3::Metadata>();
+  ON_CALL(*host, metadata()).WillByDefault(Return(metadata));
```

Comment:
How much of this common setup code can be factored out into some helper utils? E.g. `addHostWithWeightInLocalityAtPriority()`. This might make the test more readable and allow more complex examples. For example, I think covering a multiplicity of localities and priorities would be a good idea.

Reply:
Added a general helper method `addHostInfo()` to be used in all the tests for dumping eds. For this test case, modified it to two priorities, one with two localities, the other with one locality.